### DISCUSSION TITLE
Syntax file format

### DISCUSSION THREAD

**Discussion by lolbat** [coteditor/CotEditor#1224] [Q&A] [answered]
What format does CoTEditor use for its syntax files? I wanted to add a new one and was looking for some docs or samples.

Thanks

**Comment by 1024jp** [comment, marked as answer]
CotEditor currently uses an original format for syntax definition. You can edit (or add) them from CotEditor's Preferences > Format > Available Syntax Styles list.
For documentation, see CotEditor's help.
In addition, since the syntax definition file is YAML format, you can make your definition using plain text editor by following the structure of the existing syntax definition files.

<img width="703" alt="Screen Shot 2021-11-12 at 13 17 11" src="https://user-images.githubusercontent.com/1165044/141408623-fba1800d-0319-44f5-b667-bdb82a06b943.png">

**Reply by lolbat** [reply]
Thanks!

**Reply by lolbat** [reply]
Where do the syntax files reside?

**Reply by lolbat** [reply]
I just grabbed one of the files from here to use as a template

**Reply by Berovowilare** [reply]
Hello, 
Question: From where you have grabbed a file?
Thanks for answering.

**Reply by 1024jp** [reply]
@Berovowilare You can find bundled syntax definition files here: https://github.com/coteditor/CotEditor/tree/main/CotEditor/Resources/Syntaxes

**Reply by karamfd** [reply]
I added the jsonl file extension as part of json. Do I have to do this again when a new version of CotEditor is released?

**Reply by 1024jp** [reply]
@karamfd  No, you don't need it. However, it means that you made a custom JSON syntax overwriting the bundled JSON syntax in your local storage. So, future updates to the bundled JSON syntax are not applied automatically.

**Reply by karamfd** [reply]
@1024jp In that case, should I submit a pull request to add the .jsonl file extension as a part of json?

Edit: I could edit the [json.yml](https://github.com/coteditor/CotEditor/blob/main/CotEditor/Resources/Syntaxes/JSON.yml) file and add jsonl so it will be recognized as a regular json file in the future.

**Reply by 1024jp** [reply]
@karamfd Thank you for the offer. But rather than bothering you, I've just added the .jsonl extension to the bundled JSON syntax definition to the next versoin at 2532c2241390117b1acf0432fc2d05ab66d95ce9 😉.
